lib/howl/commands/app_commands.moon

```diff
@@ -302,36 +313,54 @@ command.register
 -----------------------------------------------------------------------
 
 launch_cmd = (working_directory, cmd) ->
-  shell = howl.sys.env.SHELL or '/bin/sh'
-  p = Process {
-    :cmd,
-    :shell,
-    read_stdout: true,
-    read_stderr: true,
-    working_directory: working_directory,
-  }
-
-  buffer = ProcessBuffer p
-  editor = app\add_buffer buffer
-  editor.cursor\eof!
-  buffer\pump!
+  dispatch.launch ->
```

Comment:
So, I think I like the last proposed solution the best. Since we've moved back to allowing direct dispatch anyway, we currently need to check whether arguments are provided or not anyway in the handlers. This way, the `handler` can always assume arguments are provided properly, so we would be moving that conditional stuff clearly into a separate get-the-input function. 

I'm not a big fan of any solution that would require all commands to invoke some special dispatch function - the above seems much better to me.